npm/private/utils.bzl

```diff
@@ -501,9 +501,6 @@ def _convert_v9_packages(packages, snapshots):
         # package_data can have the resolved "version" for things like https:// deps
         friendly_version = package_data["version"] if "version" in package_data else static_key[version_index + 1:]
 
-        # Convert the package_key to a rules_js compatible format
-        package_key = _to_package_key(name, version)
```

Comment:
pnpm9 aliases such as `my-name: http://with/other/name` result in a package key such as `real-name@http://with/other/name`.

In this case `_to_package_key` checking if the first character is a digit and prepending the `my-name` was the bug.  With pnpm9 the `package_key` is already correct here and we don't need anything (where v5-6 still have this).